src/runtime_src/hip/api/hip_memory.cpp

```diff
@@ -326,3 +352,31 @@ hipMemcpyHtoDAsync(hipDeviceptr_t dst, void* src, size_t size, hipStream_t strea
 {
   return handle_hip_memory_error([&] { xrt::core::hip::hip_memcpy_host2device_async(dst, src, size, stream); });
 }
+
+// Fills the first sizeBytes bytes of the memory area pointed to by dev with the constant byte value value.
+hipError_t
+hipMemsetAsync(void* dst, int value, size_t size, hipStream_t stream)
+{
+  return handle_hip_memory_error([&] { xrt::core::hip::hip_memset_async<std::uint8_t>(dst, value, size, stream); });
+}
+
+// Fills the memory area pointed to by dev with the constant integer value for specified number of times.
+hipError_t
+hipMemsetD32Async(void* dst, int value, size_t count, hipStream_t stream)
+{
+  return handle_hip_memory_error([&] { xrt::core::hip::hip_memset_async<std::uint32_t>(dst, value, count*sizeof(std::uint32_t), stream); });
+}
+
+// Fills the memory area pointed to by dev with the constant integer value for specified number of times.
+hipError_t
+hipMemsetD16Async(void* dst, int value, size_t count, hipStream_t stream)
```

Comment:
According to the spec the second argument should be `unsigned short value`

Reply:
fixed